quant: rename _ObserverBase to UniformQuantizationObserverBase

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#76461 quant: rename _ObserverBase to UniformQuantizationObserverBase**
* #76460 quant: make RecordingObserver inherit from ObserverBase

Summary:

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test plan:

CI

Differential Revision: [D35976350](https://our.internmc.facebook.com/intern/diff/D35976350)

vkuzo committed:
quant: rename _ObserverBase to UniformQuantizationObserverBase

Summary:

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test plan:

CI

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35976350).

jerryzh168 commented on torch/ao/quantization/observer.py:
```diff
@@ -173,7 +174,7 @@ def __init__(
     ) -> None:
         factory_kwargs = torch.nn.factory_kwargs(factory_kwargs)
-        super(_ObserverBase, self).__init__(dtype=dtype)
+        super(UniformQuantizationObserverBase, self).__init__(dtype=dtype)
```
this can be `super().__init__(dtype=dtype)` now I think

vkuzo replied:
that's true, I just did a string replace without trying to change anything else (to minimize time + risk), let me know if you think this additional change is worth adding to this PR - I'm flexible

jerryzh168 replied:
sounds good, I don't have a strong preference, we can do this later

jerryzh168 commented on torch/ao/quantization/observer.py:
```diff
@@ -334,7 +335,12 @@ def reset_min_max_vals(self):
 
-class MinMaxObserver(_ObserverBase):
+# Originally, this class was called `_ObserverBase`.  Keeping the old name around
+# for backwards compatibility.
```
should we add a TODO to remove it after certain release like 1.13?

vkuzo replied:
sure

vkuzo committed:
Update on "quant: rename _ObserverBase to UniformQuantizationObserverBase"


Summary:

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test plan:

CI

Differential Revision: [D35976350](https://our.internmc.facebook.com/intern/diff/D35976350)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: rename _ObserverBase to UniformQuantizationObserverBase"


Summary:

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test plan:

CI

Differential Revision: [D35976350](https://our.internmc.facebook.com/intern/diff/D35976350)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35976350).
